modules/javafx.controls/src/main/java/javafx/scene/control/SkinBase.java

```diff
@@ -238,7 +241,79 @@ protected final void registerChangeListener(ObservableValue<?> property, Consume
         return lambdaChangeListenerHandler.unregisterChangeListeners(property);
     }
 
+    /**
+     * Subclasses can invoke this method to register that they want to listen to
+     * invalidation events for the given observable. Registered {@link Consumer} instances
+     * will be executed in the order in which they are registered.
+     *
+     * @param observable the observable to observe for invalidation events
+     * @param consumer the consumer
+     * @since 17
+     */
+    protected final void registerInvalidationListener(Observable observable, Consumer<Observable> consumer) {
+        if (lambdaChangeListenerHandler == null) {
+            lambdaChangeListenerHandler = new LambdaMultiplePropertyChangeListenerHandler();
+        }
+        lambdaChangeListenerHandler.registerInvalidationListener(observable, consumer);
+    }
+
+    /**
+     * Unregisters all invalidation listeners that have been registered using
+     * {@link #registerInvalidationListener(Observable, Consumer)}
+     * for the given observable. The end result is that the given observable is no longer observed by any of the invalidation
+     * listeners, but it may still have additional listeners registered on it through means outside of
+     * {@link #registerInvalidationListener(Observable, Consumer)}.
+     *
+     * @param observable The observable for which all listeners should be removed.
```

Comment:
Maybe not "all listeners", but "the listeners". Will be less confusing and in line with the method description.

Also, the style is to use lowercase for the first word and no period at the end.